src/main/java/javaserver/ResponseBuilderFactory.java

```diff
@@ -0,0 +1,58 @@
+package javaserver;
+
+import javaserver.RequestHandlers.*;
+import javaserver.ResponseBuilders.*;
+
+public class ResponseBuilderFactory {
+
+	public static ResponseBuilder createResponder(Request request) {
+		String requestURI = getURI(request);
+		ResponseBuilder responseBuilder;
+		String requestHandlerType = Routes.acceptableRoutes.get(requestURI);
+
+		if (requestHandlerType == null) {
+			requestHandlerType = "Default Handler";
+		}
+
+		switch (requestHandlerType) {
+		case "Parameter Handler":
```

Comment:
This is a violation of OCP that is very fragile. Instead of defining your routes as a map of string keys and values, why not make the value the ResponseBuilder itself? All of the RequestHandlers appear to just override the method to get the supported Http methods, which could live inside the ResponseBuilders themselves. In that way, you could eliminate all of the handlers and this switch statement entirely, and get the ResponseBuilder from your Routes map by matching on the request's URI.